src/libstd/rt.rs

```diff
@@ -62,7 +62,7 @@ fn lang_start<T: ::termination::Termination + 'static>
             ::sys_common::backtrace::__rust_begin_short_backtrace(move || main().report())
         });
         #[cfg(not(feature = "backtrace"))]
-        let exit_code = panic::catch_unwind(mem::transmute::<_, fn()>(main).report());
+        let exit_code = panic::catch_unwind(|| main().report());
```

Comment:
Could you change this to be a move closure? This should at least be a micro-optimization